app/models/services/base-service.ts

```diff
@@ -160,7 +160,7 @@ export default abstract class BaseService<ServiceParamType> {
       do {
         // Sleep and poll for completion.  We could also use SNS or similar for a faster response
         await new Promise((resolve) => setTimeout(resolve, env.syncRequestPollIntervalMs));
-        job = await Job.byRequestId(db, requestId);
+        ({ job } = await Job.byRequestId(db, requestId));
```

Comment:
Can you explain this syntax? It looks like you are destructuring and calling the result as function?

Reply:
The parentheses are necessary to prevent the compiler from interpreting the destructuring brackets as a block. This is apparently the only way to use object destructuring of a variable that was previously declared with a `let`. You don't need them if you are declaring the variable at the same time as the destructuring, e.g., 
```
const { job } = await Job.byRequestId(db, requestId);
```
TypeScript weirdness.